channels/security/websocket.py

```diff
@@ -40,9 +54,88 @@ def valid_origin(self, origin):
         # None is not allowed
         if origin is None:
             return False
+        # Get only hostname all allowed origins
+        # Only if not full: https://domain.example.com:8443 -> domain.example.com
+        #   or //domain.example.com - > domain.example.com
+        #   or .example.com -> .example.com
+        self.allowed_origins = [urlparse(pattern).hostname or urlparse("//" + pattern).hostname
```

Comment:
Can you rewrite this without using `or`? Two in a row makes it harder to skim over.